StyleCop.Analyzers/StyleCop.Analyzers.Test/SpacingRules/SA1003UnitTests.cs

```diff
@@ -0,0 +1,469 @@
+        var v7 = ( (byte)v1 == 0);
+        var v8 = v6[ ~v1];
+    }
+}
+";
+
+            // The comments cannot be fixed, as they should not be removed automatically and moving them will cause issues in more complex statements
+            var fixedTestCode = @"public class Foo
+{
+    public void Bar()
+    {
+        for (var i = 0; i < 2; ++i)
+        {
+        }
+
+        var v1 = 1;
+        var v2 = ~v1;
+        var v3 = ~v1;
+        var v4 = ~//comment
+v1;
+        var v5 = ~/*comment*/v1;
+        var v6 = new[] { 1, 2, 3 };
+        var v7 = ((byte)v1 == 0);
+        var v8 = v6[~v1];
+    }
+}
+";
+            DiagnosticResult[] expected =
+            {
+                this.CSharpDiagnostic().WithLocation(5, 31).WithArguments("++", BePrecededBySpace),
+                this.CSharpDiagnostic().WithLocation(10, 18).WithArguments("~", NotFollowedBySpace),
+                this.CSharpDiagnostic().WithLocation(11, 18).WithArguments("~", NotAppearAtLineEnd),
+                this.CSharpDiagnostic().WithLocation(13, 18).WithArguments("~", NotFollowedByComment),
+                this.CSharpDiagnostic().WithLocation(15, 18).WithArguments("~", NotFollowedByComment),
+                this.CSharpDiagnostic().WithLocation(17, 20).WithArguments("(byte)", NotBePrecededBySpace),
+                this.CSharpDiagnostic().WithLocation(18, 22).WithArguments("~", NotBePrecededBySpace)
+            };
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, expected, CancellationToken.None);
+            await this.VerifyCSharpDiagnosticAsync(fixedTestCode, EmptyDiagnosticResults, CancellationToken.None);
+            await this.VerifyCSharpFixAsync(testCode, fixedTestCode);
+        }
+
+        /// <summary>
+        /// Verifies that valid binary expressions do not produce diagnostics.
+        /// </summary>
+        [Fact]
+        public async Task TestValidBinaryExpressions()
+        {
+            var testCode = @"using System;
+
+public class Foo
+{
+    public void Bar()
+    {
+        var v1 = 1 + 2;
+        var v2 = v1 > 0 ? 1 : 2;
+        var v3 = (2 * 3);
+        var v4 = new[] { 1, 2, 3 };
+        var v5 = 1;
+        var v6 = v4[v5 + 1];
+        Action<int> v7 = t => { v5 += v6 + t; };
+    }
+}
+";
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None);
+        }
+
+        /// <summary>
+        /// Verifies that invalid binary expressions produce the correct diagnostics and code fixes.
+        /// </summary>
+        [Fact]
+        public async Task TestInvalidBinaryExpressions()
+        {
+            var testCode = @"using System;
+
+public class Foo
+{
+    public void Bar()
+    {
+        // invalid assignment
+        var v1= 1;
+        var v2 =2;
+        var v3=3;
+
+        // invalid operators
+        var v4 = v1+ v2;
+        var v5 = v1 |v2;
+        var v6 = v1>v2;
+
+        // invalid conditionals
+        var v7 = v6? 1 : 0;
```

Comment:
:bulb: These are technically ternary operators. It would make sense to place them in a separate test.